my-vault/internal/handlers/handlers.go

```diff
@@ -0,0 +1,205 @@
+	"github.com/bmasaiti/go-projects/my-vault/internal/storage"
+	"github.com/google/uuid"
+)
+
+var logger = slog.New(slog.NewJSONHandler(os.Stdout, nil))
+
+// var db = storage.NewInMemorySecretRepo()
+// return domain.Secret{}, fmt.Errorf("secret with ID '%s' not found", secretId)
+// fmt.Errorf("no secrets found in the secrets store")
+// var ErrNotFound = errors.New("not found")
+type SecretsRepository interface {
+	PutNewSecret(secret domain.Secret) error
+	GetScretsById(Id string) (domain.Secret, error)
+	DeleteSecretByID(Id string) (string, error)
+	ListAllSecrets() ([]domain.Secret, error)
+}
+
+// Couldn't mock the tests , need to inject the db
+type SecretHandler struct {
+	DB SecretsRepository
+}
+
+type CreateSecretRequest struct {
+	Name  string            `json:"name"`
+	KVMap map[string]string `json:"kv_map"`
+}
+
+type CreateSecretResponse struct {
+	Message string `json:"message"`
+	Id      string `json:"id"`
+	Name    string `json:"name"`
+}
+type GetSecretResponse struct {
+	Message string            `json:"message"`
+	Id      string            `json:"id"`
+	Name    string            `json:"name"`
+	KVMap   map[string]string `json:"kv_map"`
+}
+
+type ListSecretsResponse struct {
+	Message string          `json:"message"`
+	Secrets []domain.Secret `json:"Secrets"`
+}
+
+func GenerateUUID() (string, error) {
+	newUUID, err := uuid.NewV7()
+	if err != nil {
+		return "", err
+	}
+	return newUUID.String(), nil
+}
+
+func NewSecretResponseObject(s domain.Secret) GetSecretResponse {
+	return GetSecretResponse{
+		Message: "Retrieved secret object",
+		Id:      s.Id,
+		Name:    s.Name,
+		KVMap:   s.KVMap,
+	}
+}
+
+func NewCreateSecret(s CreateSecretRequest) (domain.Secret, error) {
+	uuid, err := GenerateUUID()
+	if err != nil {
+		return domain.Secret{}, err
+
+	}
+
+	return domain.Secret{
+		Id:    uuid,
+		Name:  s.Name,
+		KVMap: s.KVMap,
+	}, nil
+}
+
+func BuildListSecretsResponse(s []domain.Secret) ListSecretsResponse {
+	return ListSecretsResponse{
+		Message: "Secrets fetched successfully",
+		Secrets: s,
+	}
+}
+
+func (h *SecretHandler) HandlePostSecret(w http.ResponseWriter, r *http.Request) {
+
+	//var secret Secret
+	var secretRequestObject CreateSecretRequest
+
+	err := json.NewDecoder(r.Body).Decode(&secretRequestObject)
+	defer r.Body.Close()
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		logger.Error("Unexpected internal error", "error", err)
+		return
+	}
+	// build new secretObject
+	temp, err := NewCreateSecret(secretRequestObject)
+	if err != nil {
+		secretErr := errors.New("unexpected internal error")
+		http.Error(w, secretErr.Error(), http.StatusInternalServerError)
+		logger.Error("Unexpected internal error", "error", err)
+		return
+	}
+	//err = db.PutNewSecret(temp)
+	err = h.DB.PutNewSecret(temp)
+
+	if err != nil {
+		err := errors.New("unexpected internal error")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
This is overwriting the error from the DB
```suggestion
		newErr := errors.New("unexpected internal error")
		http.Error(w, newErr.Error(), http.StatusInternalServerError)
```